src/main.py

```diff
@@ -164,14 +182,14 @@ async def main() -> None:
     """
     is_simulation = os.environ.get('OUTECH_SIMULATION',
                                    'true').lower() == 'true'
-    i = _get_container(is_simulation)
+    i = await _get_container(is_simulation)
 
     lidar_adapter: LIDARAdapter = i.get('lidar_adapter')
     lidar_controller: LidarController = i.get('lidar_controller')
     lidar_adapter.register_handler(lidar_controller.set_detection)
 
     can_adapter: CANAdapter = i.get('can_adapter')
-    socket_can_adapter: SocketCANAdapter = i.get('socket_can_adapter')
+    socket_can_adapter: SocketAdapter = i.get('socket_can_adapter')
```

Comment:
If you also use it for the lidar don't call it socket_can_adapter.

Reply:
Agreed & done